docs/kubernetes/user-guides/snapshots.md

```diff
@@ -122,3 +125,153 @@
+         resources:
+           requests:
+             storage: 6Gi
+         dataSource:
+           kind: VolumeSnapshot
+           name: restored-snapshot
+           apiGroup: snapshot.storage.k8s.io
+     EOF
+     ```
+  1. Finally, create a pod referring to the `PersistentVolumeClaim`. The PD CSI
+     driver will provision a `PersistentVolume` and populate it from the
+     snapshot. The pod from `examples/kubernetes/snapshot/restored-pod.yaml` is
+     set to use the PVC created above. After `kubectl apply`'ing the pod, run
+     `kubectl exec restored-pod -- ls -l /demo/data/` to confirm that the
+     snapshot has been restored correctly.
+
+#### Troubleshooting
+
+If the `VolumeSnapshot`, `VolumeSnapshotContents` and `PersistentVolumeClaim`
+are not all mutually synchronized, the pod will not start. Try `kubectl describe
+volumesnapshotcontent restored-snapshot-content` to see any error messages
+relating to the binding together of the `VolumeSnapshot` and the
+`VolumeSnapshotContents`. Further errors may be found in the snapshot controller
+logs:
+
+```console
+kubectl logs snapshot-controller-0 | tail
+```
+
+Any errors in `kubectl describe pvc restored-pvc` may also shed light on any
+troubles. Sometimes there is quite a wait to dynamically provision the
+`PersistentVolume` for the PVC; the following command will give more
+information.
+
+```console
+kubectl logs -n gce-pd-csi-driver csi-gce-pd-controller-0 -c csi-provisioner | tail
+```
+
+### Tips on Migrating from Alpha Snapshots
+
+The api version has changed between the alpha and beta releases of the CSI
+snapshotter. In many cases, snapshots created with the alpha will not be able to
+be used with a beta driver. In some cases, they can be migrated.
+
+>**Attention:** You may lose all data in your snapshot!
+
+>**Attention:** These instructions not work in your case and could delete all data in your
+snapshot!
+
+These instructions happened to work with a particular GKE configuration. Because
+of the variety of alpha deployments, it is not possible to verify these steps
+will work all of the time. The following has been tested with a GKE cluster of
+master version 1.17.5-gke.0 using the following image versions for the
+snapshot-controller and PD CSI driver:
+
+  * quay.io/k8scsi/snapshot-controller:v2.0.1
+  * gke.gcr.io/csi-provisioner:v1.5.0-gke.0
+  * gke.gcr.io/csi-attacher:v2.1.1-gke.0
+  * gke.gcr.io/csi-resizer:v0.4.0-gke.0
+  * gke.gcr.io/csi-snapshotter:v2.1.1-gke.0
+  * gke.gcr.io/gcp-compute-persistent-disk-csi-driver:v0.7.0-gke.0
+
+#### Migrating by Restoring from a Manually Provisioned Snapshot
+
+The idea is to provision a beta `VolumeSnapshotContents` in your new or upgraded
+beta PD CSI driver cluster with the PD snapshot handle from the alpha snapshot,
+then bind that to a new beta `VolumeSnapshot`.
+
+Note that if using the same upgraded cluster where the alpha driver was
+installed, the alpha driver must be completely removed, including its CRDs.
```

Comment:
Would it be useful to note somewhere here that uninstalling the CRD will delete all the CR objects, so it's best to save all the snapshot handles before doing that?

Reply:
I guess you also should save the information about what workload it was associated with too

Reply:
How does that sound?

FWIW I wasn't actually able to uninstall the driver & CRDs without wedging things. That's why I added the advice about spinning up a new cluster.

Reply:
If you weren't able to get it to work, then maybe let's just say in the docs that you'll need to migrate to a new cluster. The advice about saving the objects from the old cluster still stand.

Reply:
OK, I've made it slightly stronger.